contracts/epochs/EpochClusterQCs.cdc

```diff
@@ -0,0 +1,201 @@
+
+// Manages the process of collecting votes for the root quorum certificate of the upcoming
+// epoch for all collection node clusters assigned for the upcoming epoch.
+//
+// This contract is a member of a series of epoch smart contracts which coordinates the 
+// process of transitioning between epochs in Flow.
+pub contract EpochClusterQCs {
+
+    // ================================================================================
+    // CONTRACT VARIABLES
+    // ================================================================================
+
+    // Indicates whether votes are currently being collected.
+    // If false, no node operator will be able to submit votes
+    pub var inProgress: Bool
+
+    // The collection node clusters for the current epoch
+    access(account) var clusters: [Cluster]
+
+    // Votes that nodes claim at the beginning of each EpochSetup phase
+    access(account) var generatedVotes: {String: Vote}
+
+    // Votes submitted per cluster
+    access(account) var votesByCluster: {UInt16: [Vote]}
+
+    // Indicates if a voter resource has already been claimed by a node ID
```

Comment:
Adding a note that this design requires certain properties be true of node IDs, which aren't currently enforced. Need to document and enforce these on the protocol side and ideally have node IDs assigned by the smart contract during the staking process. 

I think we should aim for these properties:
* a node has the same node ID for the duration of their continual participation in the network (until they complete the un-staking process)
* node IDs are never re-used

Reply:
These are currently enforced by the identity table smart contract. 👍